libs/utils/analysis/latency_analysis.py

```diff
@@ -762,21 +762,21 @@ def _getTaskData(self, task):
                                       ','.join(self._trace.getTaskByPid(pid)))
                 self._log.warning('Returning stats only for PID: %d',
                                   task_pid)
-            task_names = self._trace.getTaskByPid(task_pid)
+            task_name = self._trace.getTaskByPid(task_pid)
 
         # Get task name
         elif isinstance(task, int):
             task_pid = task
-            task_names = self._trace.getTaskByPid(task_pid)
-            if len(task_names) == 0:
+            task_name = self._trace.getTaskByPid(task_pid)
+            if len(task_name) == 0:
```

Comment:
I've just spotted that this is wrong too - `getTaskByPid` returns `None` when it isn't found (it used to return `[]`). Could you change this to `if task_name is None` as well? Sorry I didn't notice this before.

Reply:
Yes... this has changed when I've added the patch to always return the lest known name for each PID... before we used to return a list of names in case the task was changing its name after fork. ;-)

Thanks @Leo-Yan for the fix, it was in my todo list since a while +1